### PR TITLE
[WEB-3926] feat: port underlying Accordion to radix, add product icons, menu styles, bump to 14.8.0

### PR DESCRIPTION
**Pull request by jamiehenson**
## Jira Ticket Link / Motivation

Related to https://ably.atlassian.net/browse/WEB-3926, provides things the new docs nav needs.

### Radix
We don't need a dedicated `Menu` component or anything like that, we can repurpose our existing `Accordion` component. In this PR, I've switched the underlying foundation of `Accordion` to use Radix's Accordion [component](https://www.radix-ui.com/primitives/docs/components/accordion) - purely to eliminate the maintenance burden of keeping up with the fundamentals when there is already a fully formed, light primitive already out there.

Going forward, I will investigate Radix alternatives for other components - the Tooltip is also a good candidate.

### Accordion props
Additionally, I've added a few icons that the new docs nav needs, and the following props to the `options` block, each of which adds further styling to the section the names depict:
- `headerCSS`
- `selectedHeaderCSS`
- `contentCSS`

### Other
I've also added TypeDoc references to Accordion types so that helpful descriptions can be surfaced downstream in tooltips in IDEs - I saw that Radix does this and thought it was a neat convention to adopt with our stuff as well.

Furthermore, I've added classes to represent the menu typography styling in the design system (i.e. `ui-menu-label-4`).

## Merge/Deploy Checklist

- [ ] Written automated tests for implemented features/fixed bugs
- [ ] Rebased and squashed commits
- [ ] Commits have clear descriptions of their changes
- [ ] Checked for any performance regressions

## Frontend Checklist

- [ ] No frontend changes in this PR
- [ ] Added before/after screenshots for changes
- [ ] Tested on different platforms/browsers with [Browserstack](https://www.browserstack.com)
- [ ] Compared with the initial design / our [brand guidelines](https://www.figma.com/file/jTQgyIovrhGBOf4Zrvjvbk/Ably-Linear-Design?node-id=118%3A0)
- [ ] Checked the code for accessibility issues ([VoiceOver User Guide](https://support.apple.com/en-gb/guide/voiceover/welcome/mac))?

## Release Notes

- **New Features**
  - Introduced a new Accordion component utilizing Radix UI for improved functionality and performance.
  - Added customizable CSS options for Accordion headers and icons.
  - New stories for Icon component showcasing dark and light themes.

- **Bug Fixes**
  - Enhanced theme management for Accordion with clear mappings and utility functions.

- **Documentation**
  - Improved type definitions and documentation for Accordion-related types.

- **Style**
  - New CSS classes and animations added for Accordion components.
  - Introduced new font size variable for consistent typography.

- **Chores**
  - Updated package dependencies for enhanced functionality.

### CODE REVIEW

**Comment by jamiehenson** [comment]
@coderabbitai review

File: src/core/Accordion.tsx [hunk 1]
```diff
@@ -1,220 +1,200 @@
...
+        </RadixAccordion>
+      ) : (
+        <RadixAccordion
+          type="multiple"
+          defaultValue={openIndexes}
+          onValueChange={(values) => setOpenRowValues(values)}
+        >
+          {innerAccordion}
+        </RadixAccordion>
+      )}
```
[the remark this answers is not in the document]
Comment (jamiehenson) [reply src/core/Accordion.tsx:197]: I don't think this is a problem

File: package.json [hunk 2]
```diff
@@ -74,8 +74,10 @@
     "test:update-snapshots": "npx concurrently -k -s first -n \"SB,TEST\" -c \"magenta,blue\" \"yarn build-storybook && yarn http-server preview --port 6007 --silent\" \"wait-on tcp:6007 && yarn test-storybook -u --url http://127.0.0.1:6007\""
   },
   "dependencies": {
+    "@radix-ui/react-accordion": "^1.2.1",
     "addsearch-js-client": "^0.8.11",
     "array-flat-polyfill": "^1.0.1",
+    "clsx": "^2.1.1",
```
Comment (aralovelace) [inline package.json:80]: nice, I forgot about this package. and it's even in Voltaire already. will bear in mind to use this next time :D

**Review by aralovelace — APPROVED** [review]
Just had a quick peek on Radix. it looks good!  seems this is what we need to move forward especially how the Accordion works.

File: package.json (hunk 2 above)
Comment (jamiehenson) [reply package.json:80]: Yeah I forgot about it as well! Much cleaner than doing loads string interpolation as we have been doing elsewhere (and as you say we already have this package elsewhere). From what I can see `clsx` and `classnames` are basically interchangeable, so we need one or other other (though I understand that `clsx` is just the same but lighter)

**Pull request merged**